### PR TITLE
Update VS classes __str__/__repr__

### PR DESCRIPTION
**Pull request by Setsugennoao**
Added missing members, or whole repr in the case of AudioFrame, also made all consistent between each other.

### CODE REVIEW

**Comment by cid-chan** [comment]
Well. Since you made the PR, I have to break the bad news to you: I always wanted to switch to a single line-repr since this is what most people expect.

File: src/cython/vapoursynth.pyx [hunk 1]
```diff
@@ -2435,6 +2459,9 @@ cdef class _CoreProxy(object):
         
     def __setattr__(self, name, value):
         setattr(self.core, name, value)
+
+    def __str__(self):
```
Comment (cid-chan) [inline src/cython/vapoursynth.pyx:2463]: We need a case if there is no environment. Otherwise `__str__` crashes.

**Review by cid-chan — CHANGES_REQUESTED** [review]

**Comment by myrsloik** [comment]
I kinda miss the num channels bit in AudioNode string representation. The rest looks ok apart from cid's comment.

**Comment by Setsugennoao** [comment]
This now depends on #919

**Comment by Setsugennoao** [comment]
To add `Plugin` and `Function`'s `__repr__`, I opened #920 to add `version`/`bound` parameters.

**Comment by Setsugennoao** [comment]
After using this branch for a couple of days, I'm _really_ preferring the repr over the str...
At this point I'd just delete the old __str\_\_ formatting and only leave the new __repr\_\_, lol.

File: src/cython/vapoursynth.pyx [hunk 2]
```diff
@@ -1225,6 +1267,47 @@ cdef FrameProps createFrameProps(RawFrame f):
 Mapping.register(FrameProps)
 
 
+cdef class ChannelLayout(int):
+    cdef readonly int num_channels
```
Comment (cid-chan) [inline src/cython/vapoursynth.pyx:1271]: Not required.

File: src/cython/vapoursynth.pyx [hunk 3]
```diff
@@ -1225,6 +1267,47 @@ cdef FrameProps createFrameProps(RawFrame f):
...
+            if ((1 << v) & self):
+                yield v
+
+    def __eq__(self, other):
+        if not isinstance(other, ChannelLayout):
+            return False
+        return other == self
+
+    def __len__(self):
+        return self.num_channels
```
Comment (cid-chan) [inline src/cython/vapoursynth.pyx:1290]: I think, `self.bit_count()` would be another solution that doesn't require `num_channels`.

**Review by cid-chan — CHANGES_REQUESTED** [review]

File: src/cython/vapoursynth.pyx (hunk 2 above)
Comment (Setsugennoao) [reply src/cython/vapoursynth.pyx:1271]: All right, so we only use the channel layout and can put this on public API

File: src/cython/vapoursynth.pyx (hunk 3 above)
Comment (Setsugennoao) [reply src/cython/vapoursynth.pyx:1290]: All right, so we only use the channel layout and can put this on public API

**Pull request merged**